hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/validators/AddressBookValidator.java

```diff
@@ -105,14 +124,58 @@ public void validateGossipEndpoint(
      */
     public void validateServiceEndpoint(
             @Nullable final List<ServiceEndpoint> endpointList, @NonNull final NodesConfig nodesConfig) {
+        requireNonNull(nodesConfig);
+
         validateFalse(endpointList == null || endpointList.isEmpty(), INVALID_SERVICE_ENDPOINT);
         validateFalse(endpointList.size() > nodesConfig.maxServiceEndpoint(), INVALID_SERVICE_ENDPOINT);
         for (final var endpoint : endpointList) {
             validateEndpoint(endpoint, nodesConfig);
         }
     }
 
+    /**
+     * Validates the admin key.
+     *
+     * @param key The key to validate
+     * @throws PreCheckException if the key is invalid
+     */
+    public void validateAdminKey(@Nullable Key key) throws PreCheckException {
+        final var keyEmpty = isEmpty(key);
+        validateFalsePreCheck(key == null || keyEmpty, KEY_REQUIRED);
+        validateTruePreCheck(isValid(key), INVALID_ADMIN_KEY);
+    }
```

Comment:
Why do we need a different method for this ? `isEmpty` is already checked in `isValid` . We could just use isValid and null without this method

Reply:
Removed null check. I wanted to return a different error code when key is empty and key is invalid. Added this method to take care of both.